softlayer/resource_softlayer_scale_group.go

```diff
@@ -245,9 +245,8 @@ func buildScaleVlansFromResourceData(d *schema.Set, meta interface{}) ([]datatyp
 
 		if len(networkVlans) < 1 {
 			return nil, fmt.Errorf(
-				"Unable to locate a vlan matching the provided router hostname and vlan number: %s/%s",
-				primaryRouterHostname,
-				vlanNumber)
+				"Unable to locate the specified vlan: %d.%s",
+				vlanNumber, primaryRouterHostname)
```

Comment:
Ok.  I see your point now.  It's not very clear as-is either.

I believe this is somewhat moot right now anyway, as we were planning to re-work this schema to accept a VLAN ID (once the new datasource is in-place).  So this method will probably be moved to the new vlan datasource anyway?  That said, if we really want to eliminate all ambiguity, then I guess the message would need to look something like:
"Invalid vlan: the combination of vlan number %d and hostname %s do not identify an existing vlan"